### PR TITLE
API-enabled BookInfo sample application

### PR DESCRIPTION
**Pull request by ErikWittern**
This pull request adds a JSON API to the BookInfo application. Changed details, ratings, and reviews microservices now return JSON responses. The productpage microservice fetches JSON responses and either renders them in HTML productpage, or exposes them via API endpoints. Added a `swagger.yaml` file documenting the API. Exposed the API via ingress controller through changes in `bookinfo.yaml`.

The motivations behind this pull request are to more closely follow best practices of microservices communicating in a lean, machine-understandable data format, and to allow experiments with an API as discussed with @rshriram.

```release-note
Updated bookinfo services to support proper REST responses (JSON format). You can now use swagger UI to access bookinfo API on your cluster.
```

### CODE REVIEW

**Comment by ErikWittern** [comment]
I signed it!

**Comment by ErikWittern** [comment]
@frankbu Good point! I have committed a change in 04ee5fab838d786e29b024592c9fd0f1835c7e1c to display nothing if no ratings are present, rather than displaying an error-message.

**Comment by rshriram** [comment]
```release-note
Updated bookinfo app with proper REST style API and Swagger files
```

**Comment by rshriram** [comment]
All necessary tests pass again.. Phew! merging..

File: samples/apps/bookinfo/src/ratings/ratings.js [hunk 1]
```diff
@@ -12,134 +12,85 @@
...
-var http = require('http');
-var dispatcher = require('httpdispatcher');
+var http = require('http')
+var dispatcher = require('httpdispatcher')
 
-port = parseInt(process.argv[2]);
+var port = parseInt(process.argv[2])
 
-if (process.env.SERVICE_VERSION == "v2") {
-  if (process.env.DB_TYPE == "mysql") {
```
Comment (ldemailly) [inline samples/apps/bookinfo/src/ratings/ratings.js]: please restore this ASAP

**Pull request merged**